src/middleware/mongodb.js

```diff
@@ -1,14 +1,65 @@
 import mongoose from 'mongoose';
 import {CONNECT_MONGO_URL} from "../../config";
+import {log} from '../utils/utils';
 
-const connectDB = (handler) => async (req, res) => {
+const onMongooseConnectionError = (error) => {
+  log(`Runtime mongo error: ${error}`);
+};
+
+const connectDB = async (callback) => {
   if (mongoose.connections[0].readyState) {
-    // Use current db connection
-    return handler(req, res);
+    await callback();
+  } else {
+    try {
+      await mongoose.connect(CONNECT_MONGO_URL);
+      mongoose.connection.on(`error`, onMongooseConnectionError);
+
+      await callback();
+    } catch (error) {
+      log(error);
+      let commonError = new Error();
+      error.name = `Mongodb error`;
+      error.message = `Error first connection mongo`;
+      throw commonError;
+    }
   }
-  // Use new db connection
-  await mongoose.connect(CONNECT_MONGO_URL);
-  return handler(req, res);
 };
 
-export default connectDB;
+/**
+ * @param {mongoose.Model} model
+ * @return {Object} return {ok: true|false, message: String}
```

Comment:
```suggestion
 * @return {{ok: boolean, message: string}}
```

Вот так можно это выразить через jsdoc